Modules/System/Geolocation/src/GeolocationImpl.Codeunit.al

```diff
@@ -0,0 +1,141 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+codeunit 50101 "Geolocation Impl."
+{
+    Access = Internal;
+
+    var
+        GeolocationPage: Page Geolocation;
+        LocationNotRetrievedError: Label 'The geographical location data was not retrieved.';
+        CachedLocation: Dotnet Location;
+        LocationProvider: DotNet LocationProvider;
+        LocationOptions: DotNet LocationOptions;
+        LocationOptionsEnabled: Boolean;
+
+    procedure RequestGeolocation(GeolocationImpl: Codeunit "Geolocation Impl."): Boolean
```

Comment:
What is this shenanigan here? 😄

Reply:
The Geolocation page needs a reference to the GeolocationImpl, in order to call `LocationInteractionOnOpenPage` and `LocationInteractionOnLocationAvailable`. This was the best solution I could come up with to ensure that the Page can call those functions on the GeolocationImpl object. If we had a `this` keyword the parameter could have been avoided.

Reply:
Is this because `LocationProvider` _must_ be on the page, but it's also part of the codeunit's state? 🤔 

Well, if it works as it is, let's not try to make it even more complicated.